lib/heidrun/mapping_tools/marc.rb

```diff
@@ -0,0 +1,209 @@
+module Heidrun
+  module MappingTools
+    ##
+    # Static methods for evaluations and assignments that can be shared between
+    # mappings of MARC providers.
+    #
+    module MARC
+      require_relative 'marc/genre'
+      require_relative 'marc/dctype'
+      require_relative 'marc/dcformat'
+
+      module_function
+
+      ##
+      # Given options representing leader and controlfield values, return an
+      # array of applicable genre (edm:hasType) controlled-vocabulary terms
+      #
+      # Options are as follows, and are all Strings:
+      #   - leader:  value of MARC 'leader' element
+      #   - cf_007:  value of MARC 'controlfield' with 'tag' attribute '007'
+      #   - cf_008:  value of MARC 'controlfield' with 'tag' attribute '008'
+      #   - cf_970a:  TODO
+      #
+      # @param opts [Hash] Options, as outlined above
+      # @return [Array]
+      def genre(opts)
+        genres = []
+        args = [genres, opts]
+        # TODO: insert handling of optional cf_970a for Hathi, as the first
+        # evaluation.
+        Genre.assign_language(*args) || Genre.assign_musical_score(*args) \
+          || Genre.assign_manuscript(*args) || Genre.assign_maps(*args) \
+          || Genre.assign_projected(*args) || Genre.assign_two_d(*args) \
+          || Genre.assign_nonmusical_sound(*args) \
+          || Genre.assign_musical_sound(*args)
+        genres << 'Government Document' \
+          if Genre.government_document?(opts[:cf_008])
+        genres
+      end
+
+      def dctype(opts)
+        types = []
+        args = [types, opts]
+        DCType.assign_337a(*args)
+        DCType.assign_text(*args) \
```

Comment:
I see your point about top-to-bottom scanning; still it caught my eye as busy and hard to read. :) 

It's also off-guide ([#no-trailing-backslash](https://github.com/bbatsov/ruby-style-guide#no-trailing-backslash)), so if we go this way we should document it.